Fleet UI: Side bar restyled,  update UI for Team/Roles to match Figma

RachelElysia opened the pull request:
- Revamp the sidebars in Fleet UI to match Figma
- Added user roles and teams to settings sidebar
- Changed colors of some roles / teams to match Figma
- Updated adminUserStub and userStub to match an admin API keys: `global_role: admin; teams: []` and a user API
 `global_role: null; teams: [{teamStub}]`
- Replaced most `userStub` tests with `adminUserStub` because that's what we're testing the most on jest tests
- 
Closes #1608 
Closes #1656 

Just a heads up that I followed User Settings in Figma which had more horizontal padding (40px).
<img width="283" alt="Screen Shot 2021-08-12 at 2 32 28 PM" src="https://user-images.githubusercontent.com/71795832/129250041-280772f8-718c-42ec-a7b8-9d02e73a3b96.png">

Old loom 8/12:
https://www.loom.com/share/cd528a0f27164aa788db296020ce5fc6
New loom 8/13:
https://www.loom.com/share/134df583cbd7429183e5c08f5413573f

noahtalerman reviewed (changes requested):
Sweet! @RachelElysia can you please increase the sidebar width on the "Manage hosts" page to `340px` as well? 

Since we want this width to be consistent for all sidebars, I believe the best way to make this change is to edit `$sidepanel-width` and `$sidepanel-tablet-width` in 9frontend/styles/var/_global.scss](https://github.com/fleetdm/fleet/blob/main/frontend/styles/var/_global.scss) file. Let's make them both `340px`. Thanks!

RachelElysia commented on frontend/components/side_panels/HostSidePanel/PanelGroupItem/PanelGroupItem.jsx:
```diff
@@ -56,7 +79,10 @@ class PanelGroupItem extends Component {
         <div className={`${baseClass}__flexy`}>
-          <span className={`${baseClass}__name`}>{displayText}</span>
+          <span className={`${baseClass}__name`}>
+            {label_type === "builtin" && displayIcon(name)}&nbsp;
```
I know @gillespi314 and I prefer `{" "}` but it might be good to specify not to wrap between icon and a text with `&nbsp;`? Idk @zwass -- any overarching opinion?

zwass replied:
I wasn't familiar with this so had a quick look. It makes sense to me to use a "non-breaking space" in a place we don't want any wrapping, such as between the icon and text like you mention.

Unless I'm missing something, my preference would be to use `&nbsp;` where the non-breaking behavior is desired and `{" "}` where breaking is okay.

RachelElysia commented:
@noahtalerman - Ok, I cleaned everything up and this is ready for a final review.

I added Role/Team into this PR instead of the other because it was in sidebar and a part of styling (also looked weird to grey out font when the background was still grey in the other PR).

https://www.loom.com/share/134df583cbd7429183e5c08f5413573f

noahtalerman reviewed (approved):
So good!

> I added "Tables" to match what's displayed in Figma

^You're making it a good Friday. I appreciate your ability to notice small inconsistencies in the live app and Figma

And I appreciate the Loom walkthrough as always 😄

zwass commented on frontend/fleet/helpers.ts:
```diff
@@ -410,6 +412,75 @@ export const formatPackForClient = (pack: any): any => {
+  globalRole: string | null
+): string => {
+  if (globalRole === null) {
+    const listOfRoles: any = teams.map((team) => team.role);
```
Maybe this?

```suggestion
    const listOfRoles: []string = teams.map((team) => team.role);
```

zwass commented on frontend/fleet/helpers.ts:
```diff
@@ -410,6 +412,75 @@ export const formatPackForClient = (pack: any): any => {
+    } else if (teams.length === 1) {
+      // no global role and only one team
+      return stringUtils.capitalize(teams[0].role ?? "");
+    } else if (
+      listOfRoles.every((role: string): boolean => role === "maintainer")
+    ) {
+      // only team maintainers
+      return stringUtils.capitalize(teams[0].role ?? "");
+    } else if (
+      listOfRoles.every((role: string): boolean => role === "observer")
+    ) {
+      // only team observers
+      return stringUtils.capitalize(teams[0].role ?? "");
+    }
```
Can this be simplified?

```suggestion
    } else if (
      listOfRoles.every((role: string): boolean => role === "maintainer")
    ) {
      // only team maintainers
      return "Maintainer"
    } else if (
      listOfRoles.every((role: string): boolean => role === "observer")
    ) {
      // only team observers
      return "Observer"
    }
```

RachelElysia replied:
Good catch, I made old code a helpers function but I didn't consider cleaning it up! 😅

zwass reviewed (approved):
Looks good!

A couple of code style suggestions.
